clients/src/main/java/org/apache/kafka/common/requests/ProduceRequest.java

```diff
@@ -225,33 +214,28 @@ public void clearPartitionRecords() {
         data = null;
     }
 
+    //FIXME Check this logic
```

Comment:
Is this still valid?

Reply:
This seems fine, I'll just remove the FIXME.